use_cases/eluc/app/app.py

```diff
@@ -1,118 +1,49 @@
-from app.components.sliders import SlidersComponent
-from app.components.trivia import TriviaComponent
+from app.utils import EvolutionHandler
+
 
 app = Dash(__name__,
            external_stylesheets=[dbc.themes.BOOTSTRAP, dbc.icons.BOOTSTRAP],
            prevent_initial_callbacks="initial_duplicate")
 server = app.server
 
-df = pd.read_csv(app_constants.DATA_FILE_PATH, index_col=app_constants.INDEX_COLS)
-
-legend_component = LegendComponent()
-
-map_component = MapComponent(df)
-map_component.register_update_map_callback(app)
-map_component.register_click_map_callback(app)
-map_component.register_select_country_callback(app)
-
-prescription_component = PrescriptionComponent(df)
-prescription_component.register_select_prescriptor_callback(app)
-prescription_component.register_toggle_modal_callback(app)
-
-sliders_component = SlidersComponent(df)
-sliders_component.register_set_frozen_reset_sliders_callback(app)
-sliders_component.register_show_slider_value_callback(app)
-sliders_component.register_sum_to_one_callback(app)
-
-lock_component = LockComponent()
-
-chart_component = ChartComponent(df)
-chart_component.register_update_context_chart_callback(app)
-chart_component.register_update_presc_chart_callback(app)
+app.title = 'Land Use Optimization'
 
-prediction_component = PredictionComponent(df)
-prediction_component.register_predictor_callback(app)
-prediction_component.register_land_use_callback(app)
+app_df = pd.read_csv("app/data/app_data.csv", index_col=app_constants.INDEX_COLS)
 
-trivia_component = TriviaComponent(df)
-trivia_component.register_update_trivia_callback(app)
+evolution_handler = EvolutionHandler()
 
+intro_component = IntroComponent()
+context_component = ContextComponent(app_df, evolution_handler)
+filter_component = FilterComponent(evolution_handler)
+dms_component = DMSComponent(app_df, evolution_handler)
 references_component = ReferencesComponent()
 
-app.title = 'Land Use Optimization'
-app.css.config.serve_locally = False
-# Don't be afraid of the 3rd party URLs: chriddyp is the author of Dash!
-# These two allow us to dim the screen while loading.
-# See discussion with Dash devs here: https://community.plotly.com/t/dash-loading-states/5687
-app.css.append_css({'external_url': 'https://codepen.io/chriddyp/pen/bWLwgP.css'})
-app.css.append_css({'external_url': 'https://codepen.io/chriddyp/pen/brPBPO.css'})
-
-app.layout = html.Div([
-    dcc.Markdown('''
-# Land Use Optimization
-This site is for demonstration purposes only.
-
-For a given context cell representing a portion of the earth,
-identified by its latitude and longitude coordinates, and a given year:
-* What changes can we make to the land usage
-* In order to minimize the resulting estimated CO2 emissions? (Emissions from Land Use Change, ELUC,
- in tons of carbon per hectare)
+context_component.register_callbacks(app)
+filter_component.register_callbacks(app)
+dms_component.register_callbacks(app)
```

Comment:
Instead of all the long series of callbacks we now register all the callbacks at once with these simple functions. We could abstract these into a general `Component` class later and just call `register_callbacks` on them all.